### ISSUE TITLE
Add .text-gray-dark utility class

### ISSUE THREAD

**Issue by eddyerburgh** [twbs/bootstrap#19842]
Bootstrap needs a .text-gray-dark utility class.

Currently .bg-danger on an ancestor will set all child paragraph text font color to #fff.

A .text-gray-dark utility  would be useful for a situation where you wanted to override an ancestor .bg- class changing color to white

**Comment by RyanZim** [comment]
v3 is closed to new features, but :+1: for adding v4.

**Comment by eddyerburgh** [comment]
This was meant for v4. Sorry, I'm having trouble forking and adding issues to v4. Seems the README hasn't been updated.

**Comment by praveenpuglia** [comment]
In my case with v4, I too needed class like that and other grays and I ended up creating a **_utils-extended.scss** in my styles directory and used the `@include text-emphasis-variant()` mixin as follows.

```scss
@include text-emphasis-variant('.text-gray-dark', #373737);
@include text-emphasis-variant('.text-gray', #707276);
@include text-emphasis-variant('.text-gray-light', #999);
```

I am not sure if those should be available by default in Bootstrap though. I created an issue about the `bg-*` classes here where the text color was hard coded to `#fff`. #19854

**Comment by eddyerburgh** [comment]
@praveenpuglia see #19847 . It only includes 1 extra class and sits nicely with the other `.text-color `utilities. 

I think reverting the color of text that has a `bg-color` ancestor will be a common need.

I'm gonna update my pull request next week when I've got some time. The new class will be `.text-color-default`

Thoughts?

**Comment by praveenpuglia** [comment]
Based on the convention followed in creating other emphasis classes e.g 
```
$brand-primary -> .text-primary
```
It makes sense to name it `.text-gray-dark` but that would confuse people in thinking that there might be similar classes available for *gray* or *gray-light* which is bad.

That way I feel `.text-color-default` sounds good.

**Comment by Igcorreia** [comment]
![image](https://cloud.githubusercontent.com/assets/1076409/17583937/a5dc31c2-5fab-11e6-95b3-e6da6778be5a.png)

-> http://v4-alpha.getbootstrap.com/components/utilities/

**Comment by RyanZim** [comment]
@Igcorreia That is a separate issue. It looks like it's been fixed in commit https://github.com/twbs/bootstrap/commit/7d2295ffaf3a2833f47c164539872de5d2f63b15.

**Comment by mdo** [comment]
Merged #19847.
